app/static/voice.js

```diff
@@ -0,0 +1,68 @@
+function listVoices(synth, voiceSelect) {
+	voices = synth.getVoices();
+	const selectedIndex = Math.max(0, voiceSelect.selectedIndex);
+
+	voiceSelect.innerText = '';
+	for (const voice of voices) {
+		const option = document.createElement('option');
+		option.textContent = `${voice.name} (${voice.lang})`;
+		if (voice.default) {
+			option.textContent += ' -- DEFAULT';
+			option.selected = true;
+		}
+		option.setAttribute('data-lang', voice.lang);
+		option.setAttribute('data-name', voice.name);
+		voiceSelect.appendChild(option);
+	}
+	voiceSelect.selectedIndex = selectedIndex;
+}
+
+
+function setOutput(ctrl) {
+	const range = document.getElementById(ctrl);
+	range.nextElementSibling.textContent = range.value;
+}
+
+
+window.addEventListener("load", (event) => {
```

Comment:
This function is too complicated - can you please split it into multiple functions?